cram_pr2/cram_pr2_projection/src/ik.lisp

```diff
@@ -154,82 +153,85 @@
-                                          :torso-upper-limit torso-upper-limit
-                                          :test-angle next-test-angle)
-                       (when ik-solution
-                         (values ik-solution result-angle))))
-                   (format t "IK service call freaked out. No IK solution found.~%")))))
-      (roslisp:set-debug-level NIL old-debug-lvl))))
+  (let ((ik-link (get-ik-solver-link left-or-right)))
+    (handler-case
+        (roslisp:with-fields ((response-error-code (val error_code))
+                              (joint-state (joint_state solution)))
+            (progn
+              (roslisp:wait-for-service (concatenate 'string
+                                                     (getf *ik-service-namespaces* left-or-right)
+                                                     "/get_ik_solver_info") 10.0)
+              (roslisp:call-service
+               (concatenate 'string (getf *ik-service-namespaces* left-or-right) "/get_ik")
+               "moveit_msgs/GetPositionIK"
+               (roslisp:make-request
+                "moveit_msgs/GetPositionIK"
+                (:ik_link_name :ik_request) ik-link
+                (:pose_stamped :ik_request) (cl-transforms-stamped:to-msg
+                                             (cram-tf:ensure-pose-in-frame
+                                              cartesian-pose
+                                              cram-tf:*robot-torso-frame*
+                                              :use-zero-time T))
+                (:joint_state :robot_state :ik_request) (or seed-state
+                                                            (make-zero-seed-state left-or-right))
+                (:timeout :ik_request) 1.0)))
+          (cond ((eql response-error-code
+                      (roslisp-msg-protocol:symbol-code
+                       'moveit_msgs-msg:moveiterrorcodes
+                       :success)) joint-state)
+                ((eql response-error-code
+                      (roslisp-msg-protocol:symbol-code
+                       'moveit_msgs-msg:moveiterrorcodes
+                       :no_ik_solution)) nil)
+                (T (error 'simple-error
+                          :format-control "IK service failed: ~a"
+                          :format-arguments (list
+                                             (roslisp-msg-protocol:code-symbol
+                                              'moveit_msgs-msg:moveiterrorcodes
+                                              response-error-code))))))
+      (simple-error (e)
+        (declare (ignore e))
+        nil))))
 
+(defparameter *torso-step* 0.01)
 
+(defun call-ik-service-with-torso-resampling (left-or-right cartesian-pose
+                                              &key seed-state test-angle torso-angle torso-lower-limit torso-upper-limit)
+  (labels ((_call-ik-service-with-torso-resampling (left-or-right cartesian-pose
```

Comment:
Thanks for providing the PR!

Could you please rename ``_call-ik-service-with-torso-resampling`` into ``call-ik-service-with-torso-resampling-inner`` or so? Underscores are very awkward in Lisp, usually inner functions of recursion are named with ``inner`` suffix.

Reply:
Done